src/xlsx/conversion/data_validation_conversion.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  XLSX_DV_DATE_OPERATOR_TO_DV_TYPE_MAPPING,
+  XLSX_DV_DECIMAL_OPERATOR_TO_DV_TYPE_MAPPING,
+} from "./conversion_maps";
+
+export function convertDataValidationRules(
+  xlsxDataValidations: XLSXDataValidation[],
+  warningManager: XLSXImportWarningManager
+): DataValidationRuleData[] {
+  const dvRules: DataValidationRuleData[] = [];
+  let dvId = 1;
+  for (const dv of xlsxDataValidations) {
+    if (!dv) {
+      continue;
+    }
+    switch (dv.type) {
+      case "time":
+        warningManager.generateNotSupportedWarning(WarningTypes.TimeDataValidationNotSupported);
+        break;
+      case "textLength":
+        warningManager.generateNotSupportedWarning(
+          WarningTypes.TextLengthDataValidationNotSupported
+        );
+        break;
+      case "whole":
+        warningManager.generateNotSupportedWarning(
+          WarningTypes.WholeNumberDataValidationNotSupported
+        );
+        break;
+      case "decimal":
+        const decimalRule = convertDecimalRule(dvId++, dv);
+        dvRules.push(decimalRule);
+        break;
+      case "list":
+        const listRule = convertListrule(dvId++, dv);
+        dvRules.push(listRule);
+        break;
+      case "date":
+        if (dv.operator === "notEqual") {
+          warningManager.generateNotSupportedWarning(
+            WarningTypes.NotEqualDateDataValidationNotSupported
+          );
+          break;
+        }
+        const dateRule = convertDateRule(dvId++, dv);
+        dvRules.push(dateRule);
+        break;
+      case "custom":
+        const customRule = convertCustomRule(dvId++, dv);
+        dvRules.push(customRule);
+        break;
+    }
+  }
+  return dvRules;
+}
+
+function convertDecimalRule(id: number, dv: XLSXDataValidation): DataValidationRuleData {
+  const values = [dv.formula1.toString()];
+  if (dv.formula2) {
+    values.push(dv.formula2.toString());
+  }
+  return {
+    id: id.toString(),
+    ranges: dv.sqref,
+    isBlocking: dv.errorStyle !== "warning",
+    criterion: {
+      type: XLSX_DV_DECIMAL_OPERATOR_TO_DV_TYPE_MAPPING[dv.operator] ?? "isBetween",
+      values,
+    },
+  };
+}
+
+function convertListrule(id: number, dv: XLSXDataValidation): DataValidationRuleData {
+  const values = [dv.formula1.toString()];
+  if (dv.formula2) {
+    values.push(dv.formula2.toString());
+  }
+  return {
+    id: id.toString(),
+    ranges: dv.sqref,
+    isBlocking: dv.errorStyle !== "warning",
+    criterion: {
+      type: "isValueInRange",
+      values: dv.formula1.split(","),
+      displayStyle: "arrow",
+    },
+  };
+}
+
+function convertDateRule(id: number, dv: XLSXDataValidation): DataValidationRuleData {
+  const values = [dv.formula1.toString()];
+  if (dv.formula2 && dv.formula2 !== "undefined") {
+    values.push(dv.formula2.toString());
+  }
+  let criterion = {
+    type: XLSX_DV_DATE_OPERATOR_TO_DV_TYPE_MAPPING[dv.operator] ?? "dateIsBetween",
```

Comment:
Should we really have a default ? The operator not being in the mapping mean that we do  not support it, we should skip the rule conversion altogether.